src/colmap/feature/pairing_test.cc

```diff
@@ -239,6 +239,176 @@ TEST(SequentialPairGenerator, Quadratic) {
 }
 
 TEST(SpatialPairGenerator, Nominal) {
+  {
+    constexpr int kNumImages = 3;
```

Comment:
I suggest to decouple this into different `TEST` cases with descriptive names for what we test.